packages/discovery-react-components/src/components/StructuredQuery/components/RemoveRuleRowButton/RemoveRuleRowButton.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { FC, Dispatch, SetStateAction } from 'react';
+import { Button } from 'carbon-components-react';
+import SubtractAlt16 from '@carbon/icons-react/lib/subtract--alt/16';
+import { Messages } from 'components/StructuredQuery/messages';
+import {
+  StructuredQuerySelection,
+  Row
+} from 'components/StructuredQuery/utils/structuredQueryInterfaces';
+
+export interface RemoveRuleRowButtonProps {
+  /**
+   * text to display for the Remove rule button icon description
+   */
+  removeRuleRowButtonIconDescription: Messages['removeRuleRowButtonIconDescription'];
+  /**
+   * id of the group for the rule row to render, or 'top-level' if the top-level rule group
+   */
+  groupId: number | 'top-level';
+  /**
+   * id of the rule row to remove
+   */
+  rowId: Row['id'];
+  /**
+   * state that represents the current rules and selections for the structured query
+   */
+  structuredQuerySelection: StructuredQuerySelection;
+  /**
+   * used to set the structuredQuerySelection state
+   */
+  setStructuredQuerySelection: Dispatch<SetStateAction<StructuredQuerySelection>>;
+}
+
+export const RemoveRuleRowButton: FC<RemoveRuleRowButtonProps> = ({
+  removeRuleRowButtonIconDescription,
+  groupId,
+  rowId,
+  structuredQuerySelection,
+  setStructuredQuerySelection
+}) => {
+  const handleOnClick = () => {
+    if (groupId === 'top-level') {
+      setStructuredQuerySelection({
+        ...structuredQuerySelection,
+        rows: structuredQuerySelection.rows.filter((ruleRow: Row) => ruleRow.id !== rowId)
+      });
+    } else {
+      setStructuredQuerySelection({
+        ...structuredQuerySelection,
```

Comment:
something to be careful of here - this is only doing a shallow copy. i don't think there are any gotchas with this code as it stands, but could turn hard-to-debug issues later on (nested objects being mutated)